sdk/servicebus/azure-servicebus/azure/servicebus/_common/utils.py

```diff
@@ -196,6 +205,25 @@ def transform_messages_to_sendable_if_needed(messages):
         except AttributeError:
             return messages
 
+def create_messages_from_dicts_if_needed(messages, message_type):
+    """
+    This method is used to convert dict representations
+    of messages and to a list of ServiceBusMessage objects or ServiceBusBatchMessage.
+    """
+    # type: (DictMessageType) -> Union[List[azure.servicebus.ServiceBusMessage], azure.servicebus.ServiceBusMessageBatch]
+    if isinstance(messages, list):
+        for index, message in enumerate(messages):
+            if isinstance(message, dict):
+                messages[index] = message_type(**message)
```

Comment:
emmm, I don't think we should touch users' input:

```suggestion
        return [(message_type(**message) if isinstance(message, dict) else message) for message in messages]
```